python/mujoco_mpc/demos/direct_optimizer.py

```diff
@@ -0,0 +1,1479 @@
+# Copyright 2023 DeepMind Technologies Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import matplotlib.pyplot as plt
+import mujoco
+import numpy as np
+from numpy import typing as npt
+from typing import Tuple
+
+# %%
+# differentiate mj_differentiatePos
```

Comment:
@yuvaltassa add this method to MuJoCo? Here's a [C++ implementation](https://github.com/google-deepmind/mujoco_mpc/blob/15ac83b052f09faddc5e6e874e8439a0a0611c64/mjpc/utilities.cc#L1228C6-L1228C35).